numpy/_core/tests/test_ufunc.py

```diff
@@ -1743,6 +1740,15 @@ def test_where_with_broadcasting(self):
         assert_array_equal((a[where] < b_where), out[where].astype(bool))
         assert not out[~where].any()  # outside mask, out remains all 0
 
+    def test_where_warns(self):
+        a = np.arange(7)
+        mask = a % 2 == 0
+        with warnings.catch_warnings(record=True) as w_where:
```

Comment:
Please use `with pytest.warns()`, also removes the need for that `'where' in str(...)`